pkg/jobs/registry.go

```diff
@@ -186,10 +186,7 @@ type Registry struct {
 func (r *Registry) UpdateJobWithTxn(
 	ctx context.Context, jobID jobspb.JobID, txn isql.Txn, updateFunc UpdateFn,
 ) error {
-	job, err := r.LoadJobWithTxn(ctx, jobID, txn)
-	if err != nil {
-		return err
-	}
+	job := Job{registry: r, id: jobID}
```

Comment:
instantiating a job struct directly like this could be unsafe if a subsequent method on the job struct assumes the job struct contains some state. It also seems like we do this elsewhere. Yet another job todo would be to create some sort of `statelessJob` interface on which we can safely call these methods.

Reply:
Definitely agree having some different types here would be good.